src/freenet/client/filter/ElementInfo.java

```diff
@@ -218,7 +227,27 @@ public class ElementInfo {
 			"first-letter",
 			"before",
 			"after",
-			"target"
+			"target",
+			"any-link",
+			"default",
+			"defined",
+			"disabled",
```

Comment:
Why do we need to discuss where these can be used? As long as it's valid it should be here, right? And if it has problems, we can add it into the BANNED_PSEUDOCLASS. Freesites do have forms, could be Library, commentmaker or some other future plugin.

Reply:
I’m being careful with adding selectors that might not be usable, because I have to check with each element, whether it could allow sneaking in an unfiltered URL. That’s why the review of these takes a while (need to look at mdn for each).

Also we have to need tests for them all.

That said: better CSS support is great and I love it that you’re pushing the state here!

Reply:
… so yes, disabled should be useful, even when we can’t use all of the capabilities of forms.